src/runtime_src/core/pcie/driver/linux/xocl/subdev/mb_scheduler.c

```diff
@@ -2531,8 +2531,21 @@ exec_mark_cmd_state(struct exec_core *exec, struct xocl_cmd *xcmd, enum ert_cmd_
 static inline void
 exec_mark_cmd_complete(struct exec_core *exec, struct xocl_cmd *xcmd)
 {
+	//For soft kernels get cmd state and return_code
+	uint32_t cmd_state = ERT_CMD_STATE_COMPLETED;
+	if (cmd_type(xcmd) == ERT_SCU) {
+		struct xocl_ert *xert = exec_is_ert(exec) ? exec->ert : NULL;
+		if (xert) {
+			uint32_t slot_addr = xcmd->slot_idx * xert->slot_size;
+			struct ert_start_kernel_cmd *pkt = xcmd->ert_cu;
+			xocl_memcpy_fromio((void*)&pkt->return_code, xert->cq_base + slot_addr, 2 * sizeof(u32));
+			cmd_state = pkt->state;
+			pkt->return_code = pkt->data[0];
```

Comment:
Why are we using pkt->data[0] for return code other than reusing cu_exta_mask?

Reply:
cu_extra_mask is reused. Just copied 64 bits are shifted to proper area. Not overwriting full header (first 32 bits)